### DISCUSSION TITLE
prep Failed At cargo prisma "generate"

### DISCUSSION THREAD

**Discussion by bytrangle** [spacedriveapp/spacedrive#360] [Community Support] [answered]
While running `pnpm prep`, I encountered a problem similar to this #118 , but the cause is due to "no match for id ...".

Here's the whole log:

```
error: failed to get `datamodel` as a dependency of package `prisma-client-rust v0.6.0 (https://github.com/Brendonovich/prisma-client-rust.git?branch=0.6.0#f6799c68)`
    ... which satisfies git dependency `prisma-client-rust` (locked to 0.6.0) of package `sdcore v0.1.0 (/Users/calico/Documents/dev/projects/spacedrive/core)`
    ... which satisfies path dependency `sdcore` (locked to 0.1.0) of package `spacedrive v0.1.0 (/Users/calico/Documents/dev/projects/spacedrive/apps/desktop/src-tauri)`

Caused by:
  failed to load source for dependency `datamodel`

Caused by:
  Unable to update https://github.com/Brendonovich/prisma-engines?rev=5d34c9a567aaec4065a48d8cf6c2ee885e6ca281#5d34c9a5

Caused by:
  object not found - no match for id (5d34c9a567aaec4065a48d8cf6c2ee885e6ca281); class=Odb (9); code=NotFound (-3)
```
Here's more information if that helps:

```
System Version: macOS 12.5 (21G72)
Kernel Version: Darwin 21.6.
$ pnpm --version
7.1.7
$ cargo --version
cargo 1.62
$ rustc --version
rustc 1.62
```

**Comment by bytrangle** [comment, marked as answer]
`pnpm prep` ran successfully after I upgraded to the latest nightly versions of Rustc and Cargo. So I guess using a nightly Rust is critical?

**Reply by Brendonovich** [reply]
Nightly isn't required, I think it worked because using nightly caused a refetch of the latest version of the Prisma engines, which fixed the bug you experienced. I'd recommend staying on Rust v1.63 from now on.

**Comment by oscartbeaumont** [comment]
I didn't think nightly worked with Prisma Client Rust, but if it does that's great. You will definitely require a recent version of Rust to compile Spacedrive.

@Brendonovich
